python/lsst/daf/butler/registry/bridge/monolithic.py

```diff
@@ -91,30 +91,36 @@ def _makeTableSpecs(datasets: type[DatasetRecordStorageManager]) -> _TablesTuple
     # We want the dataset_location and dataset_location_trash tables
     # to have the same definition, aside from the behavior of their link
     # to the dataset table: the trash table has no foreign key constraint.
-    dataset_location_spec = ddl.TableSpec(
+    # The order of columns in dataset_location_trash is reversed, it is more
```

Comment:
Are we sure it's not fine to also reverse it for dataset_location on the basis that for a small butler this will also result in a better query plan before a sufficient number of rows exist for ANALYZE?

Reply:
Theoretically I think it is better to have datastore name as leading part of the key, but in practice I don't think we'll see any difference. With small number of datasets I would not care about performance too much, and more important question would be how frequently ANALYZE is going to run.